langchain/package.json

```diff
@@ -142,6 +143,7 @@
     "gpt-3-encoder": "^1.1.4",
     "jsonpointer": "^5.0.1",
     "p-queue": "^7.3.4",
+    "pdf-parse": "^1.1.1",
```

Comment:
assuming this is not a crazy dependency with known issues?

Reply:
Seems good to me (although last updated 4 years ago, but I guess pdfs don't change)

Reply:
And btw if we we ever want to produce one document per page in a pdf all we have to do is start using the underlying dependency (pdf.js) directly

Reply:
pdf.js is huge. I tried to use it, the code works but it has a dependency on a package called `canva` which is huge and made it impossible to use in any serverless(Vercel, AWS lambda) environment.

Reply:
If we make it a peerDependency, it means whoever is going to use this loader would suffer from the size issue still.

Reply:
Is there an alternative pdf parser that doesn't use pdf.js? Ideally I want a pure-JS one, not one with native bindings, which would bring other problems

Reply:
I don't believe this package `pdf-parse` uses `pdf.js`. In the readme, it does mention a few alternatives tho https://www.npmjs.com/package/pdf-parse

Reply:
Sorry, it uses it... I didn't see it in `package.json`, it seems that it include it in the repo instead https://gitlab.com/autokent/pdf-parse/-/tree/master/lib/pdf.js

Reply:
I think it does, but the author bundled in the dist version, so probably a lot smaller than a direct install, see https://gitlab.com/autokent/pdf-parse/-/tree/master/lib/pdf.js

Reply:
The one I had issue is https://www.npmjs.com/package/pdfjs-dist

Reply:
Ha yes exactly! In your book-gpt you didn't have an issue getting this pdf-parse one deployed to a serverless function, right?